lib/Parser/parse.cpp

```diff
@@ -5423,6 +5455,17 @@ void Parser::ParseFncFormals(ParseNodePtr pnodeFnc, ushort flags)
                     {
                         Error(ERRRestWithDefault);
                     }
+
+                    // In defer parse mode we have to flag the function node to indicate that it has default arguments
+                    // so that it will be considered for any syntax error scenario.
+                    // Also mark it before parsing the expression as it may contain functions.
+                    ParseNode* currentFncNode = GetCurrentFunctionNode();
```

Comment:
GetCurrentFunctionNode() gives either the current function node or the current deferred function node. This is needed because if the current function is deferred and has duplicate elements in strict formals then we have to throw a syntax error as soon as we get out of this loop. Marking either the current function or the current deferred function help in identifying that scenario.